src/background/background-process.js

```diff
@@ -370,21 +419,21 @@ export default class BackgroundProcess {
           this.menuItems.openPanel.enable()
           this.menuItems.info.enable()
 
-          this.updateIcon(true)
+          this.updateIcon(true, tabId)
```

Comment:
Maybe it's better to pass `tab` into an `updateIcon` instead of tabId? This way a `tab` would be a unit of tab identification. If a downstream function will need some more information about tab later (not just tabId, but a windowId, or maybe something else as well), we will not need to change a function interface.

Reply:
UpdateIcon is a simple wrap for the browser.browserAction.setIcon(params) method, that needs only a tabId, image and path data. So it needs only tabId.
I think that the more safe and more consistent way to pass the minimum data and in scalar form Tab object is a TabScript Class object that has lots of inner objects, methods and so on. 
I believe that my approach makes code more independent for changing in future and decrease data saved in closures.
What do you think about my arguments?

Reply:
I think you're right, agree with you here.